pkg/oci/pusher/pusher.go

```diff
@@ -72,11 +86,22 @@ func (p *Pusher) Push(ctx context.Context, artifactType oci.ArtifactType,
 		return nil, err
 	}
 
+	// First thing check that we do not have multiple rulesfiles.
+	if artifactType == oci.Rulesfile && len(o.Filepaths) != 1 {
```

Comment:
I would love if artifactType was used to build an `Artifact` interface; we could use here
`artifact.Validate()`
and below
`artifact.RootDesc()`

Wdyt?